MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetSource.cs

```diff
@@ -5,26 +5,35 @@
 namespace MixedRealityExtension.Assets
 {
     /// <summary>
-    /// Documents the origin of an asset
+    /// Documents the origin of an asset.
+    /// NOTE: The InternalId is not used for equality testing.
     /// </summary>
-    public struct AssetSource
+    public class AssetSource
     {
         /// <summary>
         /// The type of container the asset came from.
         /// </summary>
-        public AssetContainerType ContainerType { get; }
+        public AssetContainerType ContainerType { get; set; }
 
         /// <summary>
         /// The URL of the asset's container.
         /// </summary>
-        public Uri Uri { get; }
+        public string Uri { get; set; }
+
+        private Uri parsedUri;
+        /// <summary>
+        /// The parsed URI of the asset's container.
+        /// </summary>
+        public Uri ParsedUri => parsedUri = parsedUri ?? new Uri(Uri);
```

Comment:
>ParsedUri [](start = 19, length = 9)

Please make sure this field is excluded from JSON serialization. Because this class is part of a network payload, it is unsafe to assume it will never be sent to the app. I'd recommend not having this field and just parsing the url in LoadAssetsFromGLTF call.

Reply:
It makes sense to be able to cache the Uri result. I've excluded it.

---
In reply to: [251068734](https://github.com/Microsoft/mixed-reality-extension-unity/pull/15#discussion_r251068734) [](ancestors = 251068734)